contents/docs/contribute/developing-locally.mdx

```diff
@@ -9,264 +9,282 @@ showTitle: true
+When spinning up an instance of PostHog for development, we recommend the following configuration:
+- the external services run in Docker over `docker compose`
+- PostHog itself runs on the host (your system)
 
-### Set up databases
+This is what we'll be using in the guide below.
 
-<HiddenSection headingType='h4' title='With Docker' id='dbs-with-docker'>
+> It is also technically possible to run PostHog in Docker completely, but syncing changes is then much slower, and for development you need PostHog dependencies installed on the host anyway (such as formatting or typechecking tools).
+> The other way around – everything on the host, is not practical due to significant complexities involved in instantiating Kafka or ClickHouse from scratch.
 
-1. Make sure you have Docker and Docker Compose installed and Docker Engine is running
-2. Run:
+The instructions here assume you're running macOS. For Linux, adjust the steps as needed (e.g. use `apt` or `pacman` in place of `brew`). Windows isn't supported.
 
-```bash
-docker-compose -f docker-compose.dev.yml up -d redis db
-```
+In case some steps here have fallen out of date, please tell us about it – feel free to [submit a patch](https://github.com/PostHog/posthog.com/tree/master/contents/docs/contribute/developing-locally.mdx)!
 
-> **Friendly tip:** If you see `Error while fetching server API version: 500 Server Error for http+docker://localhost/version:` likely Docker Engine isn't running.
 
-> **Friendly tip:** You might need `sudo`, see more [manage docker as non-root](https://docs.docker.com/engine/install/linux-postinstall).
+## macOS prerequisites
 
-</HiddenSection>
+1. Install Xcode Command Line Tools if you haven't already: `xcode-select --install`.
 
-<HiddenSection headingType='h4' title='Without Docker' id='dbs-without-docker'>
+2. Install the package manager Homebrew by following the [instructions here](https://brew.sh/).
 
-1. Make sure you have [Redis installed](https://redis.io/download) and running. We recommend using version 5 or higher.
+<blockquote class="warning-note">
+    After installation, make sure to follow the instructions printed in your terminal to add Homebrew to your <code>$PATH</code>. Otherwise the command line will not know about packages installed with <code>brew</code>.
+</blockquote>
 
-    ```bash
-    # macOS (Homebrew)
-    brew install redis && brew services start redis
-    ```
+3. Install [Docker Desktop](https://www.docker.com/products/docker-desktop) and in its settings give Docker **at least 4 GB of RAM** (or 6 GB if you can afford it) and at least 4 CPU cores.
 
-2. Make sure you have [PostgreSQL installed](https://www.postgresql.org/download/) and running. We recommend using version 11 or higher. You may also try [Postgres.app](https://postgresapp.com/), but remember to follow the instructions to add `psql` to your `$PATH` if you do.
+4. Append line `127.0.0.1 kafka clickhouse` to `/etc/hosts`. You can do it in one line with:
```

Comment:
What's the point of doing this? What about using `127.0.0.1` instead of `kafka` and `clickhouse` in the env variables?

Reply:
I'd be super happy to not need this... but have no time to try all permutations of variables/hosts now. If anyone can report our setup working well on a M1 mac without these steps, either with changed envs or not, please advise.

Reply:
Let's merge with this and then we can re-evaluate.

Reply:
this is because of an implementation detail on Kafka side. Basically your local client first connects to the broker asking for members of the kafka cluster it gives you `kafka` as the hostname (because of how docker-compose is configured with that as the name of the service & how the services refer to kafka from within the compose setup). So after your client asks the broker for the connection details it tries to connect to `kafka` host from local. Having `kafka` route to local and into the compose stack is the easiest way to get around that.

Reply:
Re: ClickHouse I'm not sure why this is required - I haven't had this issue in the past. I'll look into it.